content/lessons/solid/solid-part-2.md

```diff
@@ -0,0 +1,14 @@
+---
+title: SOLID Principles Part 2
+template: lesson
+draft: false
+slug: /courses/Solid
+course: SOLID
+tags:
+- Usability
+- Maintainability
+- OOP
+description: "Put your new SOLID knowledge to the test as you rewrite principle-violating code so it is in line with the principles."
+timeToCompletion: ~2 hours
+secondaryExerciseUrl: 'https://repl.it/@wffhuergo/Part-2a#Main.java'
```

Comment:
This is great! I'm happy to see a new course that's quite a bit different from the others.

A few thoughts on the exercises:

1. It might be nice to have an explanation in the terminal after all tests have passed. The exercises can be completed with some trial-and-error, so it would be great to see a "Here's why LSP (etc.) works in the specific case you just worked through." Just a thought.
2. CostumeFinder exercise: right now there's not much discovery, the instructions say exactly which classes and methods to create. Might be good to remove a few of the explicit lines in the instructions, or making them more vague.
3. CostumeFinder exercise: it would be great to connect the costume to the actor - right now the Actor is unused after the refactor. I could see a `getCurrentCostume` method on the Actor... it might mean more refactoring by the learner, but I think it would connect the lesson together.

Reply:
Went over this offline, updates look great 👍